lib/aiken/transaction/value.ak

```diff
@@ -378,6 +378,63 @@ test flatten_with_2() {
   ) == [("a", "2"), ("b", "")]
 }
 
+/// Reduce a value into a single result
+///
+/// ```
+/// zero()
+///  |> add("a", "1", 10)
+///  |> add("b", "2", 20)
+///  |> value.reduce(v, 0, fn(_, _, quantity, acc) { acc + quantity })
+/// // 30
+/// ```
+pub fn reduce(
+  self: Value,
+  start: acc,
+  with: fn(PolicyId, AssetName, Int, acc) -> acc,
+) -> acc {
+  dict.foldr(
+    self.inner,
+    start,
+    fn(policy_id, asset_list, value) {
+      dict.foldr(asset_list, value, with(policy_id, _, _, _))
+    },
+  )
+}
+
+test reduce_1() {
+  let v =
+    zero()
+      |> add("a", "1", 10)
+      |> add("b", "2", 20)
+  let result = reduce(v, 0, fn(_, _, quantity, acc) { acc + quantity })
+  result == 30
+}
+
+test reduce_2() {
+  let v =
+    zero()
+      |> add("a", "1", 5)
+      |> add("a", "2", 15)
+      |> add("b", "", 10)
+  let result =
+    reduce(
+      v,
+      [],
+      fn(policy_id, asset_name, _, acc) { [(policy_id, asset_name), ..acc] },
+    )
+  result == [("a", "1"), ("a", "2"), ("b", "")]
+}
+
+test reduce_3() {
```

Comment:
This test is relatively redundant with the one doing sums. May you provide a different one testing, for example, reducing on an empty Value?